src/InstrumentationEngine.Attach/Program.cs

```diff
@@ -54,6 +60,57 @@ internal static int Main(string[] args)
                 return ExitCodeFailure;
             }
 
+            // Attempt to resolve fullpath
+            string configFilePath = Path.GetFullPath(args[1]);
+            if (!File.Exists(configFilePath))
+            {
+                WriteError(Invariant($"Could not find file '{configFilePath}'."));
+                return ExitCodeFailure;
+            }
+
+            #endregion
+
+            #region Parse Configuration Sources
+            InstrumentationConfigurationSources? configSources = null;
+
+            // Schema file embedded name is the same that the full type name that was generated from the schema file
+            string schemaResourceName = typeof(InstrumentationConfigurationSources).FullName + ".xsd";
+
+            // Read schema file
+            using (Stream? schemaStream = Assembly.GetExecutingAssembly().GetManifestResourceStream(schemaResourceName))
+            using (XmlReader schemaReader = XmlReader.Create(schemaStream, new XmlReaderSettings() { XmlResolver = null }))
+            {
+                // Create reader settings that conducts validation
+                XmlReaderSettings readerSettings = new XmlReaderSettings();
```

Comment:
>XmlReaderSettings [](start = 55, length = 17)

There were some security settings that should be configured for XmlReaders, such as DtdProcessing = DtdProcessing.Prohibit

Reply:
Thanks for the catch

---
In reply to: [388514665](https://github.com/microsoft/CLRInstrumentationEngine/pull/220#discussion_r388514665) [](ancestors = 388514665)